src/sagemaker_huggingface_inference_toolkit/handler_service.py

```diff
@@ -127,12 +127,13 @@ def preprocess(self, input_data, content_type):
         decoded_input_data = decoder_encoder.decode(input_data, content_type)
         return decoded_input_data
 
-    def predict(self, data):
+    def predict(self, data, model):
         """The predict handler is responsible for model predictions. Calls the `__call__` method of the provided `Pipeline`
         on decoded_input_data deserialized in input_fn. Runs prediction on GPU if is available.
         The predict handler can be overridden to implement the model inference.
         Args:
             data (dict): deserialized decoded_input_data returned by the input_fn
+            model : Model returned by "model_fn".
```

Comment:
```suggestion
            model : Model returned by the `preprocess` method or if it is a custom module `model_fn`.
```

Reply:
This could get confusing. While `input_fn` and `model_fn` can both return a `model`, we should clearly define that the `model` should be returned from the `model_fn`.

Reply:
yep but if you don't provide a custom module the `model` will be returned by the `preprocess` function.